client/landing/stepper/declarative-flow/internals/hooks/use-preload-steps.ts

```diff
@@ -0,0 +1,79 @@
+import { SiteDetails } from '@automattic/data-stores';
+import debugFactory from 'debug';
+import { useEffect } from 'react';
+import { useSelector } from 'calypso/state';
+import { isUserLoggedIn } from 'calypso/state/current-user/selectors';
+import type { Flow, StepperStep } from '../types';
+
+const debug = debugFactory( 'calypso:stepper:preloading' );
+
+async function tryPreload( step?: StepperStep, followingStep?: StepperStep ) {
+	if ( step && 'asyncComponent' in step ) {
+		debug( 'Preloading next step:', step.slug );
+
+		await step.asyncComponent();
+
+		// Flows are indeterminate, they often pick one of the two next steps based on user input, so load two steps ahead.
+		tryPreload( followingStep );
+	}
```

Comment:
I think we need to run this even if the condition above is false, as we won't run this code if a step is not async.
```suggestion
	}

	// Flows are indeterminate, they often pick one of the two next steps based on user input, so load two steps ahead.
	if ( followingStep ) {
		tryPreload( followingStep );
	}
```

Reply:
Great catch! Addressed in [`b8d56a5` (#99352)](https://github.com/Automattic/wp-calypso/pull/99352/commits/b8d56a5dd2ffa2634c53c63e1980435edd06ff0e)